File: YouboraConfigUtils/YouboraConfigManager.m
//
//  YouboraConfigMAnager.m
//  AVPlayerAdapterExample
//
//  Created by Joan on 02/05/2017.
//  Copyright © 2017 NPAW. All rights reserved.
//

#import "YouboraConfigManager.h"

#import <YouboraLib/YouboraLib.h>

@implementation YouboraConfigManager

static YBOptions * options = nil;

+ (void) save {
    if (options != nil) {
        NSData *data = [NSKeyedArchiver archivedDataWithRootObject:options];
        [[NSUserDefaults standardUserDefaults] setObject:data forKey:@"youborasettings"];
    }
}

+ (void) reset {
    options = [YBOptions new];
    [YouboraConfigManager save];
}

+ (YBOptions *) getOptions {
    
    if (options == nil) {
        NSData *data = [[NSUserDefaults standardUserDefaults] objectForKey:@"youborasettings"];
        
        if (data == nil) {
            [self reset];
        } else {
            options = [NSKeyedUnarchiver unarchiveObjectWithData:data];
        }
    }
    
    return options;
}

@end

File: YouboraConfigUtils/YouboraConfigManager.swift
//
//  YouboraConfigManager.swift
//  YouboraConfigUtils
//
//  Created by nice on 03/10/2019.
//  Copyright © 2019 NPAW. All rights reserved.
//

import Foundation
import YouboraLib

class YouboraConfigManager {
    static public var options: YBOptions?
    
    static public func save() {
        guard let options = self.options else {
            return
        }
        UserDefaults.standard.set(NSKeyedArchiver.archivedData(withRootObject: options), forKey: Constants.youborasettings)
    }
    
    static public func reset() {
        self.options = YBOptions()
        self.save()
    }
    
    static public func getOptions() -> YBOptions {
        guard let options = self.options else {
            guard let data = UserDefaults.standard.data(forKey: Constants.youborasettings) else {
                reset()
                return self.options!
            }
            self.options = NSKeyedUnarchiver.unarchiveObject(with: data) as? YBOptions
            
            return self.options!
        }
        
        return options
    }
}
